docs/design/architecture.md

```diff
@@ -2,128 +2,615 @@
+
+The following terminology is used to describe the different or
+environments (or contexts) various processes run in. It is necessary
+to study this table closely to make sense of what follows:
+
+| Environment type | Environment name | Virtualized | Containerized | Environment rootfs | Mount type | Description |
+|-|-|-|-|-|-|-|
+| Host | Host | no `[1]` | no | Host's rootfs | &lt;any&gt; | The environment provided by a standard, physical non virtualized system. |
+| VM root | Guest VM | yes | no | rootfs inside the [guest image](#guest-image) | [DAX](#dax) | The first (or top) level VM environment created on a host system. |
+| VM container root | Container | yes | yes | rootfs type requested by user ([`ubuntu` in our example](#example-command)) | [virtio FS](#virtio-fs) | The first (or top) level container environment created inside the VM. Based on the [OCI bundle](#oci-bundle). |
+
+Key:
+
+`[1]` - For simplicity, this document assumes the host environment
+        runs on physical hardware.
+
+> **Notes:**
+>
+> - The word "root" is used to mean _top level_ here in a similar
+>   manner to the term [rootfs](#root-filesystem).
+>
+> - The term "first level" prefix used above is important since it implies
+>   that it is possible to create multi level systems. However, they do
+>   not form part of a standard Kata Containers environment so will not
+>   be considered in this document.
+
+### Container creation
+
+The steps below show at a high level how a Kata Containers container is
+created using the containerd container manager:
+
+1. The user requests the creation of a container by running a command
+   like the [example command](#example-command).
+1. The container manager daemon runs a single instance of the Kata
+   [runtime](#runtime).
+1. The Kata runtime loads its [configuration file](#configuration).
+1. The container manager calls a set of shimv2 API functions on the runtime.
+1. The Kata runtime launches the configured [hypervisor](#hypervisor).
+1. The hypervisor creates and starts (_boots_) a VM using the
+   [guest assets](#guest-assets):
+
+   - The hypervisor [DAX](#dax) shares the [guest image](#guest-image)
+     into the VM to become the VM rootfs (mounted on a `/dev/pmem*` device),
+     which is known as the [VM root environment](#environments).
+   - The hypervisor mounts the [OCI bundle](#oci-bundle), using [virtio FS](#virtio-fs),
+     into a container specific directory inside the VM's rootfs.
+
+     This container specific directory will become the
+     [container rootfs](#environments), known as the
+     [container environment](#environments).
+
+1. The [agent](#agent) is started as part of the VM boot.
+
+1. The runtime calls the agent's `CreateSandbox` API to request the
+   agent create a container:
+
+   1. The agent creates a [container environment](#environments)
+      in the container specific directory that contains the
+      [container rootfs](#environments).
+
+      The container environment
```

Comment:
The way this renders it looks like it is the second step of a 3 step process and the ii. is missing in front of it.  It renders very similar to this. 

```
i. The agent creates a container environment in the container specific directory that contains the container rootfs.

The container environment hosts the workload in the container rootfs directory.

ii. The agent spawns the workload inside the container environment.
```